libs/search-widget/src/core/old-stores/viewer.store.ts

```diff
@@ -398,17 +403,41 @@ function getParagraph(
   fieldId: string,
   field: IFieldData,
   paragraph: Paragraph,
+  ners?: PositionedNER[],
 ): WidgetParagraph {
+  let text = resource.getParagraphText(getFieldType(fieldType), fieldId, paragraph) || '';
+  if (ners) {
+    let lastNer = -1;
+    const paragraphNers = ners
+      .filter((ner) => !!ner.entity && ner.start > (paragraph.start || 0) && ner.start < (paragraph.end || 0))
+      .sort((a, b) => a.start - b.start)
+      // overlapping ners should not happen, but at the moment the API returns some
+      .filter((ner) => {
+        const isOverlapping = ner.start < lastNer;
+        lastNer = ner.end;
+        return !isOverlapping;
+      })
+      .map((ner) => ({ ...ner, start: ner.start - (paragraph.start || 0), end: ner.end - (paragraph.start || 0) }));
+    let offset = 0;
+    paragraphNers.forEach((ner) => {
+      text =
+        text.slice(0, ner.start + offset) +
+        `<mark class="ner" family="${ner.family}">${text.slice(ner.start + offset, ner.end + offset)}</mark>` +
+        text.slice(ner.end + offset);
+      offset += 35 + ner.family.length;
```

Comment:
is 35 the length of the `mark` in the dom?

Reply:
yes that's the length of `<mark class="ner" family=""></mark>`